Mage/src/main/java/mage/Mana.java

```diff
@@ -1103,7 +1142,7 @@ public int get(final ManaType manaType) {
             case GREEN:
                 return green;
             case COLORLESS:
-                return CardUtil.overflowInc(generic, colorless);
+                return CardUtil.overflowInc(generic, colorless); // TODO: This seems like a mistake
```

Comment:
This seems like a mistake. Why is this adding generic and colourless to return the amount of colourless available.

Reply:
Yes, it can be the error (added with improved colorless mana support, see 782190bac37fa104505e73e9dfe99a6d69b03c26). As I can see the pattern (search code by `switch (manaType) {`):
- mana class uses both generic and colorless types;
- mana pool class uses only colorless type.

Current code in the mana so it must use both generic and colorless types as diff types, see other methods like `set` and `increase`.